### PR TITLE
Fix problem with jobs in hack-showcase

### PR DESCRIPTION
**Pull request by wojpon**
**Description**
During the process of connecting our repository to Prow, we found a problem with jobs triggering caused by a wrong path in field "run_if_changed". To fix this problem and allow Prow to react on every change in our repository (triggering on every PR) we were advised to change this field to "always_run". 

Changes proposed in this pull request:

- Change the way of triggering jobs for [hack-showcase](https://github.com/kyma-incubator/hack-showcase)
- Add appropriate tests for this change

**Related issue**
https://github.com/kyma-incubator/hack-showcase/issues/39

### CODE REVIEW

**Comment by wojpon** [comment]
After performing some tests and debugging procedures our commit finally passes the test here on test-infra and also the tests are triggered on our hack-showcase repository.

File: prow/jobs/incubator/hack-showcase/hack-showcase.yaml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 job_template: &job_template
   skip_report: false
-  run_if_changed: "^hack-showcase/"
+  always_run: true
```
Comment (dekiel) [inline prow/jobs/incubator/hack-showcase/hack-showcase.yaml]: You should not use always_run: True for jobs which are not required for all PRs. Please correct and use run_if_changed.

File: prow/jobs/incubator/hack-showcase/hack-showcase.yaml [hunk 2]
```diff
@@ -1,8 +1,7 @@
 job_template: &job_template
   skip_report: false
-  run_if_changed: "^hack-showcase/"
```
Comment (dekiel) [inline prow/jobs/incubator/hack-showcase/hack-showcase.yaml]: run_if_changed value is a regex which is applied against files path, so it's working recursively. Path root is your repository.

File: prow/jobs/incubator/hack-showcase/hack-showcase.yaml [hunk 3]
```diff
@@ -1,8 +1,7 @@
 job_template: &job_template
```
Comment (dekiel) [inline prow/jobs/incubator/hack-showcase/hack-showcase.yaml]: Please use meaningful names.

File: prow/jobs/incubator/hack-showcase/hack-showcase.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
-job_template: &job_template
+base_job_template: &base_job_template
   skip_report: false
-  always_run: true
+  run_if_changed: "^hack-showcase/github-connector"
```
Comment (dekiel) [inline prow/jobs/incubator/hack-showcase/hack-showcase.yaml]: hack-showcase is your repository, so it's your root in file path.
Try this one : `run_if_changed: "^github-connector"`

**Pull request merged**